File: js-archive/longest-common-prefix.js
var longestCommonPrefix = function (strs) {
  if (strs.length > 0) {
    let ans = ''
    let f = strs[0]
    for (let i = 0; i < f.length; i++) {
      let isOk = true
      for (let j = 0; j < strs.length - 1; j++) {
        if (strs[j][i] === strs[j + 1][i]) {
          isOk = true
        } else {
          isOk = false
          break
        }
      }
      if (isOk) {
        ans += f[i]
      } else {
        break
      }
    }

    return ans
  } else {
    return ""
  }
};

console.log(longestCommonPrefix(["reflower", "flow", "flight"]))

File: go-archive/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    ans:=""; min:=len(strs[0])

    for _, s:=range strs {
        if (len(s) < min) {min = len(s)}
    }

    for i:=0; i < min; i++ {
        f:=strs[0][i]; isOk := true
        for k:=1; k < len(strs); k++ {
            if (strs[k][i] != f) {
                isOk = false
                break
            }
        }

        if (isOk) {
            ans+=string(f)
        } else {
            return ans
        }
    }

    return ans
}